### PR TITLE
"Structuring Your Project"  fill in

### LINKED ISSUE #30: Structuring a Python project
This is similar to #4, but we really need a section on what a basic Python package should contain and how it should be structured.

I'm thinking we should create a minimal example, say in http://github.com/omnia-md/examplepackage, that has all of the working bits of a state-of-the-art project, but just doesn't do much.

Modeling this on [mdtraj](https://github.com/mdtraj/)---noting especially the [tools license](https://github.com/mdtraj/mdtraj/tree/master/devtools#tools-license), we might adopt:
```
devtools/
  README.md
  travis-ci/
    install_miniconda.sh
  appveyor-ci/
    run_with_env.cmd
  conda-recipe/
    meta.yaml
    build.sh
    bld.bat
    README.md
docs/
examplepackage/
  tests/
    __init__.py
    test_examplepackage.py
  __init__.py
  examplepackage.py
LICENSE.md
README.md
setup.py
basesetup.py
.travis.yml
.appveyor.yml
.gitignore
```

### PR DESCRIPTION
**Pull request by Lnaden**
This adds in the content of the structure page from #30. 

The added image is referenced by the page and once merge should render for the `license_highlight` references.

Fixes #30

### CODE REVIEW

File: STRUCTURING_YOUR_PROJECT.md [hunk 1]
```diff
@@ -1,5 +1,161 @@
...
+  appveyor-ci/
+    run_with_env.cmd
+  conda-recipe/
+    meta.yaml
+    build.sh
+    bld.bat
+    README.md
+docs/
+examplepackage/
+  tests/
```
Comment (jchodera) [inline STRUCTURING_YOUR_PROJECT.md]: Might be good to highlight where to put test data as well (`examplepackage/tests/data/` or `examplepackage/data/`?)

File: STRUCTURING_YOUR_PROJECT.md [hunk 2]
```diff
@@ -1,5 +1,161 @@
...
+  examplepackage.py
+LICENSE.md
+README.md
+setup.py
+.travis.yml
+.appveyor.yml
+.gitignore
+```
+
+This list may appear daunting at first, but each element is either a critical component of what will make the software
```
Comment (jchodera) [inline STRUCTURING_YOUR_PROJECT.md]: We could likely make it less daunting by setting up an `examplepackage` repo with all of this boilerplate already filled in. Or we could write a Python script that sets up this boilerplate for them, asking a few simple questions (or reading from a YAML file).

Comment (jchodera) [reply STRUCTURING_YOUR_PROJECT.md]: But it's still important to explain why we need all of this! So this is still super useful.

File: STRUCTURING_YOUR_PROJECT.md [hunk 3]
```diff
@@ -1,5 +1,161 @@
...
+```
+LICENSE.md
+README.md
+setup.py
+.travis.yml
+.appveyor.yml
+.gitignore
+```
+
+The root directory should contain the primary `setup.py` file for the `examplepackage`. This is the primary file that
```
Comment (jchodera) [inline STRUCTURING_YOUR_PROJECT.md]: Would be good to hyperlink things like `setup.py` to more information about what these files contain.

File: STRUCTURING_YOUR_PROJECT.md [hunk 4]
```diff
@@ -1,5 +1,161 @@
...
+.travis.yml
+.appveyor.yml
+.gitignore
+```
+
+The root directory should contain the primary `setup.py` file for the `examplepackage`. This is the primary file that 
+users and deployment tools will run to build the package. This is a critical component of any Python package and 
+should always be included.
+
+The `.travis.yml` and `.appveyor.yml` are the config files needed by Travis (Linux/OSX) and AppVeyor (Windows)
```
Comment (jchodera) [inline STRUCTURING_YOUR_PROJECT.md]: Hyperlink Travis and AppVeyor?

File: STRUCTURING_YOUR_PROJECT.md [hunk 5]
```diff
@@ -1,5 +1,161 @@
...
+are no mandatory file names, but file names should be descriptive.
+
+The `conda-recpie` directory is also an arbitrary name, but contains the files needed to do a `conda` deployment 
+on the operating systems. We have [an entire chapter for deployment][package_deploy_page] and will not go into 
+further details here.
+
+## Core package files should include tests
+
+```
+examplepackage/
```
Comment (jchodera) [inline STRUCTURING_YOUR_PROJECT.md]: Might once again suggest where to put test data.

File: STRUCTURING_YOUR_PROJECT.md (hunk 2 above)
Comment (Lnaden) [reply STRUCTURING_YOUR_PROJECT.md]: I'll make a new issue for it, we can address some question I have for it there better and adjust this page once we have the boilerplate in place.

**Comment by Lnaden** [comment]
Should be ready for one more once over, @jchodera

File: STRUCTURING_YOUR_PROJECT.md (hunk 2 above)
Comment (ctk3b) [reply STRUCTURING_YOUR_PROJECT.md]: A compchem focused [cookiecutter](https://github.com/audreyr/cookiecutter) repo would be awesome!

Comment (Lnaden) [reply STRUCTURING_YOUR_PROJECT.md]: Great idea! I have added the idea to the discussion in #54 as we figure it out.

**Pull request merged**